ocaml/xapi-cli-server/records.ml

```diff
@@ -5242,3 +5242,119 @@ let vtpm_record rpc session_id vtpm =
+            |> fun s -> if s = "" then "all" else ""
+          )
+          ()
+      ; make_field ~name:"name_label"
+          ~get:(fun () -> (x ()).API.tracing_name_label)
+          ()
+      ; make_field ~name:"tags"
+          ~get:(fun () ->
+            List.map
+              (fun (k, v) -> Printf.sprintf "%s : %s" k v)
+              (x ()).API.tracing_tags
+            |> String.concat ","
+            |> fun s -> if s = "" then "none" else s
+          )
+          ~set:(fun s ->
+            String.split_on_char ',' s
+            |> List.filter_map (fun kv ->
+                   match String.split_on_char ':' kv with
+                   | [k; _] when List.mem k ["host"; "pool"; "host name"] ->
+                       None (* Reserved Tags*)
+                   | [k; v] ->
+                       Some (k, v)
+                   | _ ->
+                       None
+               )
+            |> fun tags ->
+            Client.Tracing.set_tags ~rpc ~session_id ~self:tracing ~tags
+          )
+          ()
+      ; make_field ~name:"endpoints"
+          ~get:(fun () ->
+            (x ()).API.tracing_endpoints |> String.concat ", " |> fun s ->
+            if s = "" then "none" else s
+          )
+          ~set:(fun s ->
+            String.split_on_char ',' s |> List.filter (( <> ) "none")
+            |> fun endpoints ->
+            Client.Tracing.set_endpoints ~rpc ~session_id ~self:tracing
+              ~endpoints
+          )
+          ()
+      ; make_field ~name:"components"
+          ~get:(fun () ->
+            (x ()).API.tracing_components |> String.concat ", " |> fun s ->
+            if s = "" then "all" else s
+          )
+          ~set:(fun s ->
+            String.split_on_char ',' s |> List.filter (( <> ) "all")
+            |> fun components ->
+            Client.Tracing.set_components ~rpc ~session_id ~self:tracing
+              ~components
+          )
+          ()
+      ; make_field ~name:"filters"
+          ~get:(fun () ->
+            (x ()).API.tracing_filters |> String.concat ", " |> fun s ->
+            if s = "" then "all" else s
+          )
+          ~set:(fun s ->
+            String.split_on_char ',' s |> List.filter (( <> ) "all")
+            |> fun filters ->
+            Client.Tracing.set_filters ~rpc ~session_id ~self:tracing ~filters
+          )
+          ()
+      ; make_field ~name:"processors"
+          ~get:(fun () ->
+            (x ()).API.tracing_processors |> String.concat ", " |> fun s ->
+            if s = "" then "none" else s
+          )
+          ~set:(fun s ->
+            String.split_on_char ',' s |> List.filter (( <> ) "none")
+            |> fun processors ->
+            Client.Tracing.set_processors ~rpc ~session_id ~self:tracing
+              ~processors
+          )
+          ()
+      ; make_field ~name:"status"
+          ~get:(fun () ->
+            (x ()).API.tracing_status |> fun b ->
+            if b then "enabled" else "disabled"
```

Comment:
Is this how other booleans are implemented? Please try to follow conventions used in the records.ml module

Reply:
I think a better name for the field in the datamodel is `enabled`. A "status" of true or false doesn't mean much to me. And then in the CLI it is just `enabled=true|false`.